src/main/java/org/opensearch/neuralsearch/processor/chunker/FixedTokenLengthChunker.java

```diff
@@ -0,0 +1,174 @@
+     * 2. token_limit: the token limit for each chunked passage
+     * 3. overlap_rate: the overlapping degree for each chunked passage, indicating how many token comes from the previous passage
+     * Here are requirements for parameters:
+     * max_token_count and token_limit should be a positive integer
+     * overlap_rate should be within range [0, 0.5]
+     * tokenizer should be string
+     */
+    @Override
+    public void validateParameters(Map<String, Object> parameters) {
+        validatePositiveIntegerParameter(parameters, TOKEN_LIMIT_FIELD, DEFAULT_TOKEN_LIMIT);
+        validateDoubleParameterWithinRange(parameters, OVERLAP_RATE_FIELD, OVERLAP_RATE_LOWER_BOUND, OVERLAP_RATE_UPPER_BOUND);
+        validateStringParameter(parameters, TOKENIZER_FIELD, false);
+        String tokenizer = parseStringParameter(parameters, TOKENIZER_FIELD, DEFAULT_TOKENIZER);
+        if (!WORD_TOKENIZERS.contains(tokenizer)) {
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.ROOT,
+                    "tokenizer [%s] is not supported for [%s] algorithm. Supported tokenizers are %s",
+                    tokenizer,
+                    ALGORITHM_NAME,
+                    WORD_TOKENIZERS
+                )
+            );
+        }
+    }
+
+    /**
+     * Parse the parameters for fixed token length algorithm,
+     * will throw IllegalArgumentException when parameters are invalid
+     *
+     * @param parameters a map non-runtime parameters as the following:
+     * 1. tokenizer: the <a href="https://opensearch.org/docs/latest/analyzers/tokenizers/index/">analyzer tokenizer</a> in opensearch
+     * 2. token_limit: the token limit for each chunked passage
+     * 3. overlap_rate: the overlapping degree for each chunked passage, indicating how many token comes from the previous passage
+     */
+    @Override
+    public void parseParameters(Map<String, Object> parameters) {
+        this.tokenLimit = parseIntegerParameter(parameters, TOKEN_LIMIT_FIELD, DEFAULT_TOKEN_LIMIT);
+        this.overlapRate = parseDoubleParameter(parameters, OVERLAP_RATE_FIELD, DEFAULT_OVERLAP_RATE);
+        this.tokenizer = parseStringParameter(parameters, TOKENIZER_FIELD, DEFAULT_TOKENIZER);
+    }
+
+    /**
+     * Return the chunked passages for fixed token length algorithm
+     *
+     * @param content input string
+     * @param runtimeParameters a map for runtime parameters, containing the following runtime parameters:
+     * 1. max_token_count the max token limit for the tokenizer
+     */
+    @Override
+    public List<String> chunk(final String content, final Map<String, Object> runtimeParameters) {
+        // before chunking, validate and parse runtimeParameters
+        validatePositiveIntegerParameter(runtimeParameters, MAX_TOKEN_COUNT_FIELD, DEFAULT_MAX_TOKEN_COUNT);
+        int maxTokenCount = parseIntegerParameter(runtimeParameters, MAX_TOKEN_COUNT_FIELD, DEFAULT_MAX_TOKEN_COUNT);
```

Comment:
there should be a max value what user can enter in maxtokenCount other wise user can put very very high value and can crash the system